web_widget_text_markdown/static/src/js/web_widget_text_markdown.js

```diff
@@ -0,0 +1,159 @@
+                this.$markdown.setContent(this.value || "");
+            }.bind(this), $input);
+            return $input;
+        },
+
+        _renderReadonly: function () {
+            this.$el.html(marked(this._formatValue(this.value)));
+        },
+
+
+        _getMarkdownOptions: function () {
+            var markdownOpts = {
+                autofocus: false,
+                savable: false,
+                language: this.getSession().user_context.lang,
+            };
+
+            // Only can create attachments on non-virtual records
+            if (this.res_id) {
+                var self = this;
+                markdownOpts.dropZoneOptions = {
+                    paramName: 'ufile',
+                    url: '/web/binary/upload_attachment',
+                    acceptedFiles: 'image/*',
+                    width: 'o_field_text_markdown',
+                    params: {
+                        csrf_token: core.csrf_token,
+                        session_id: this.getSession().override_session,
+                        callback: '',
+                        model: this.model,
+                        id: this.res_id,
+                    },
+                    success: function () {
+                        self._markdownDropZoneUploadSuccess(this);
+                    },
+                    error: function () {
+                        self._markdownDropZoneUploadError(this);
+                    },
+                    init: function () {
+                        self._markdownDropZoneInit(this);
+                    },
+                };
+
+                if (_t.database.multi_lang && this.field.translate) {
+                    markdownOpts.additionalButtons = [
+                        [{
+                            name: 'oTranslate',
+                            data: [{
+                                name: 'cmdTranslate',
+                                title: _t('Translate'),
+                                icon: {glyph: 'glyphicon glyphicon-flag'},
+                                callback: this._markdownTranslate,
+                            }],
+                        }],
+                    ];
+                }
+            }
+
+            return markdownOpts;
+        },
+
+        _getAttachmentId: function (response) {
+            var matchElms = response.match(/"id":\s?(\d+)/);
+            if (matchElms && matchElms.length) {
+                return matchElms[1];
+            }
+            return null;
+        },
+
+        _markdownDropZoneInit: function (markdown) {
+            var self = this;
+            var caretPos = 0;
+            var $textarea = null;
+            markdown.on('drop', function (e) {
+                $textarea = $(e.target);
+                caretPos = $textarea.prop('selectionStart');
+            });
+            markdown.on('success', function (file, response) {
+                var text = $textarea.val();
```

Comment:
When trying to upload a file I get:
`TypeError: Cannot read property 'val' of null`

Also, it seems that not all the file types can be uploaded ex. use drag-drop on a pdf file and an error is raised `You can't upload files of this type.`. Since this an unhandled exception, the web client renders it as if a crash happened. Perhaps we can catch that and raise it diferently

Reply:
Thanks, changed to console warning... user can know that is a failed upload with the icons.